string type
---
PartInfo[] parts_info
